### DISCUSSION TITLE
Writing rust procedural macro via RustPython

### DISCUSSION THREAD

**Discussion by evan0greenup** [RustPython/RustPython#5023] [Ideas]
Python as a language has much more self-reflection ability than Rust.

Procedural Macro has a meta-programming method, If it can utilize the potential of dynamic feature of RustPython. It would be dramatic.

Because Procedural Macro is computed during compile-time, the performance drop due to the involvement of RustPython interpreter doesn't affect runtime speed.

I would like to discuss about the feasibility of this idea.

**Comment by youknowone** [comment]
Do you have a suggestion how it will look like with sample code?

**Reply by evan0greenup** [reply]
The central point here is to ensure that the macro-crate has ability to run RustPython runtime and feed the `TokenStream` as the RustPython custom representation and after processing, convert back to `TokenStream`.
